src/socket-activate/socket-activate.c

```diff
@@ -113,19 +114,24 @@ static int open_sockets(int *ret_epoll_fd) {
                 log_set_open_when_needed(false);
         }
 
-        epoll_fd = epoll_create1(EPOLL_CLOEXEC);
-        if (epoll_fd < 0)
-                return log_error_errno(errno, "Failed to create epoll object: %m");
+        if (!arg_now) {
+                epoll_fd = epoll_create1(EPOLL_CLOEXEC);
+                if (epoll_fd < 0)
+                        return log_error_errno(errno, "Failed to create epoll object: %m");
+        }
+
 
         for (int fd = SD_LISTEN_FDS_START; fd < SD_LISTEN_FDS_START + count; fd++) {
                 _cleanup_free_ char *name = NULL;
 
                 getsockname_pretty(fd, &name);
                 log_info("Listening on %s as %i.", strna(name), fd);
 
-                r = add_epoll(epoll_fd, fd);
-                if (r < 0)
-                        return r;
+                if (!arg_now) {
```

Comment:
Maybe better to change the condition to `if (epoll_fd >= 0)`

Reply:
I do not think so?
If `arg_now` is used in condition, then we need to check `epoll_fd` is valid at this stage when we read/write code. epoll_fd initialization in open_sockets() is far from run(), hence little bit hard.

Another option is adding assert in `run()`:
```c
if (!arg_now) {
        assert(epoll_fd >= 0);

        if (epoll_wait(epoll_fd, &event, 1, -1) < 0) {
                ...
}
```